pkg/service/connections.go

```diff
@@ -71,10 +77,54 @@ func (s *Service) TerminateUDPConnectionsToBackend(l3n4Addr *lb.L3n4Addr) {
 			log.Errorf("Forcefully terminating sockets connected to deleted service backends " +
 				"not supported by underlying kernel: see kube-proxy free guide for " +
 				"the required kernel configurations")
+			return
 		} else {
-			log.Errorf("Error while forcefully terminating sockets connected to"+
-				"deleted service backend: %v. If you see any traffic going to such"+
-				"deleted backends, consider restarting application pods sending the traffic.", err)
+			log.WithError(err).WithField(logfields.L3n4Addr, l3n4Addr).Error(
+				"error while forcefully terminating sockets connected to" +
+					"deleted service backend. Consider restarting any application pods sending traffic" +
+					"to the backend")
+		}
+	}
+
+	// Iterate over all pod network namespaces, and terminate any stale connections.
+	if option.Config.EnableSocketLBPodConnectionTermination && !option.Config.BPFSocketLBHostnsOnly {
+		files, err := os.ReadDir(defaults.NetNsPath)
+		if err != nil {
+			log.WithError(err).WithFields(logrus.Fields{
+				"netns-dir":        defaults.NetNsPath,
+				logfields.L3n4Addr: l3n4Addr,
+			}).Error("Error opening the netns dir while " +
+				"terminating connections to deleted service backend")
+			return
+		}
+
+		for _, file := range files {
+			ns, err := netns.OpenPinned(filepath.Join(defaults.NetNsPath, file.Name()))
+			if err != nil {
+				log.WithError(err).WithFields(logrus.Fields{
+					"netns": file.Name(),
+				}).Debug("Error opening netns")
+				continue
+			}
+			err = ns.Do(func() error {
+				return s.backendConnectionHandler.Destroy(sockets.SocketFilter{
```

Comment:
iiuc, this should also take the socket cookie into account so we don't destroy anything unrelated

Reply:
>If there are netns that have nothing to do with pods, we iterate those netns too, 

AFAIK, non-pod network namespaces are not mounted under the path we are iterating by container runtimes. 

>but we don't destroy unrelated sockets, right? Because we check cilium_lb[4,6]_reverse_sk before destroying sockets.

Yes, that's right.